packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -73,4 +73,69 @@ export class ApplicationExtension<
     public beforeRouteMatch(routes: RouteProps[]): RouteProps[] {
         return routes
     }
+
+    public async serialize(): Promise<SerializedRoute[]> {
```

Comment:
This method being async is also suspicious. The flow that I was anticipating was to have the ssr flow load the routes and those routes get "cached" in a private variable that the serializer can use.